test/components/generators/test_hugging_face_local_generator.py

```diff
@@ -428,32 +405,51 @@ def test_warm_up_set_stopping_criteria_list(
         """
-        Test that warm_up method sets the `stopping_criteria_list` attribute
-        if `stop_words` is provided
+        Test that warm_up method sets the `stopping_criteria_list` attribute if `stop_words` is provided
         """
         generator = HuggingFaceLocalGenerator(
-            model="google/flan-t5-base", task="text2text-generation", stop_words=["coca", "cola"]
+            model="google/flan-t5-small", task="text2text-generation", stop_words=["coca", "cola"]
         )
-
         generator.warm_up()
-
         stop_words_criteria_mock.assert_called_once()
         stopping_criteria_list_mock.assert_called_once()
-
         assert hasattr(generator, "stopping_criteria_list")
 
     def test_run_stop_words_removal(self):
-        """
-        Test that stop words are removed from the generated text
-        (does not test stopping text generation)
-        """
+        """Test that stop words are removed from the generated text (does not test stopping text generation)"""
         generator = HuggingFaceLocalGenerator(
-            model="google/flan-t5-base", task="text2text-generation", stop_words=["world"]
+            model="google/flan-t5-small", task="text2text-generation", stop_words=["world"]
         )
-
-        # create the pipeline object (simulating the warm_up)
         generator.pipeline = Mock(return_value=[{"generated_text": "Hello world"}])
-
         results = generator.run(prompt="irrelevant")
-
         assert results == {"replies": ["Hello"]}
+
+    @pytest.mark.integration
+    def test_stop_words_criteria_using_hf_tokenizer(self):
+        """
+        Test that StopWordsCriteria will caught stop word tokens in a continuous and sequential order in the input_ids
+        using a real Huggingface tokenizer.
+        """
+        from transformers import AutoTokenizer
+
+        model_name = "google/flan-t5-small"
+        tokenizer = AutoTokenizer.from_pretrained(model_name)
+        criteria = StopWordsCriteria(tokenizer=tokenizer, stop_words=["unambiguously"])
+
+        text_one = "This is ambiguously, but is unrelated."
+        generated_text_ids = tokenizer.encode(text_one, add_special_tokens=False, return_tensors="pt")
+        assert criteria(generated_text_ids, scores=None) is False
+
+        text_two = "This is unambiguously"
+        generated_text_ids = tokenizer.encode(text_two, add_special_tokens=False, return_tensors="pt")
+        assert criteria(generated_text_ids, scores=None) is True
+
+    @pytest.mark.integration
+    def test_hf_pipeline_runs_with_our_criteria(self):
+        """Test that creating our own StopWordsCriteria and passing it to a Huggingface pipeline works."""
+        generator = HuggingFaceLocalGenerator(
+            model="google/flan-t5-small", task="text2text-generation", stop_words=["unambiguously"]
+        )
+        generator.warm_up()
+        results = generator.run(prompt="something that triggers something")
+        assert results["replies"] != []
```

Comment:
maybe we can also check that `generator.stopping_criteria_list is not None`.

Reply:
yes, makes sense! 👍🏽